lib/docgen/class_template.md.erb

```diff
@@ -0,0 +1,81 @@
+<%= kind %> <%= title %>
+========================
+<%= comment %>
+<% if constants.any? -%>
+- [Constants](#constants)
+<%- end -%>
+<%- if attributes.any? -%>
+- [Attributes](#attributes)
+<%- end -%>
+<%- if class_methods.any? -%>
+- [Class Methods](#class-methods)
+<%- class_methods.each do |method| -%>
+  - [<%= method.title %>](#<%= method.title.gsub("?", "") %>)
+<%- end -%>
+<%- end -%>
+<%- if instance_methods.any? -%>
+- [Instance Methods](#instance-methods)
+<%- instance_methods.each do |method| -%>
+  - [<%= method.title %>](#<%= method.title.gsub("?", "") %>)
+<%- end -%>
+<%- end -%>
+<%- if included.any? -%>
+- [Includes](#includes)
+<%- end -%>
+<%- if extended.any? -%>
+- [Extends](#extends)
+<%- end -%>
+<%- if constants.any? %>
+# Constants
```

Comment:
I would make this and the other in-page headings below into `h2` so they are smaller than the page title, and the hierarchy of information is clearer.

```suggestion
## Constants
```